azkaban-common/src/main/java/azkaban/utils/ExternalLinkUtils.java

```diff
@@ -100,4 +133,65 @@ static String encodeToUTF8(final String url) {
     }
     return "";
   }
+
+  /**
+   * Return list of external analyzers based on configs and current web request to web server
+   * to render the execution page.
+   * @param azkProps azkaban configs
+   * @param req current web request
+   * @return list of external analyzers
+   */
+  public static List<ExternalLink> getExternalAnalyzers( final Props azkProps, final HttpServletRequest req) {
+    final List<String> externalTopics = getExternalAnalyzerTopics(azkProps);
+    return externalTopics.stream().map(topic -> {
+      String externalLinkUrl = getExternalAnalyzerLinkOnReq(topic, azkProps, req);
+      if (!externalLinkUrl.isEmpty()) {
+        logger.debug("Adding an External analyzer to the page");
+        final String execExternalLinkLabel = getExternalLabelForTopic(topic, azkProps);
+        logger.debug("External analyzer label set to : " + execExternalLinkLabel);
+        int timeout =
+            azkProps.getInt(Constants.ConfigurationKeys.AZKABAN_SERVER_EXTERNAL_ANALYZER_TIMEOUT_MS,
+            DEFAULT_AZKABAN_SERVER_EXTERNAL_TIMEOUT_MS);
+        return new ExternalLink(topic, execExternalLinkLabel, externalLinkUrl,
+            validExternalLink(externalLinkUrl, timeout));
+      }
+      return null;
+    }).filter(Objects::nonNull).collect(Collectors.toList());
+  }
+
+  /**
+   * check if external link is reachable or not.
+   * It will send http request to target url, and check the status code from response,
+   * If the page returns 200x or 300x status, returns true, and external link is enabled,
+   * otherwise returns false, and external link is disabled.
+   * @param externalLinkUrl azkaban configs
+   * @param timeout timeout for httpClient in milliseconds
+   * @return true if target link is reachable, otherwise false.
+   */
+  private static boolean validExternalLink(final String externalLinkUrl, final int timeout) {
+    final RequestConfig config = RequestConfig.custom()
+        .setConnectTimeout(timeout)
+        .setConnectionRequestTimeout(timeout)
+        .setSocketTimeout(timeout).build();
+    final CloseableHttpClient client = HttpClients.custom()
+        .setDefaultRequestConfig(config)
+        .setSSLSocketFactory(null).disableRedirectHandling().build();
+    try {
+      HttpResponse httpResponse = client.execute(new HttpGet(externalLinkUrl));
+      int responseCode = httpResponse.getStatusLine().getStatusCode();
+      if (responseCode >= HttpStatus.SC_BAD_REQUEST) {
+        logger.warn("validExternalLink url " + externalLinkUrl + " not reachable, response code " + responseCode);
+        return false;
+      }
+      logger.debug("validExternalLink url " + externalLinkUrl + " is reachable.");
+      client.close();
```

Comment:
Could you add `client.close()` to a finally clause instead to make sure it will always be closed? Currently if an exception occurs this line might not be executed.

Reply:
Thanks, done.